13_07_2017/3D_slab_modes.py

```diff
@@ -0,0 +1,779 @@
+                        # Boundary color and other options
+                        boundary_r.module_manager.scalar_lut_manager.lut.table = lut
+                        boundary_l.module_manager.scalar_lut_manager.lut.table = lut
+                        boundary_r.actor.property.lighting = False
+                        boundary_r.actor.property.shading = False
+                        boundary_l.actor.property.lighting = False
+                        boundary_l.actor.property.shading = False
+                                                     
+                    else:
+                        #remove previous boundaries
+                        if t != 0 or view_ind != 0:
+                            boundary_r.remove()
+                            boundary_l.remove()
+                            
+                        # Make a fading colormap by changing opacity at ends
+                        lut = np.reshape(np.array([150, 150, 150, 255]*256), (256,4))
+                        fade_value = 20
+                        lut[:fade_value,-1] = np.linspace(0, 255, fade_value)
+                        lut[-fade_value:,-1] = np.linspace(255, 0, fade_value)
+                        
+                        # Set up boundary visualisation
+                        boundary_r = mlab.mesh(xix_boundary_r_vals_t, zgrid_zy, ygrid_zy,
+                                               extent=[ext_min_r, ext_max_r, 1, nz, 0, (ny-1) * y_spacing],
+                                               opacity=0.7, scalars=zgrid_zy)
+        
+                        boundary_l = mlab.mesh(xix_boundary_l_vals_t, zgrid_zy, ygrid_zy,
+                                               extent=[ext_min_l, ext_max_l, 1, nz, 0, (ny-1) * y_spacing],
+                                               opacity=0.7, scalars=zgrid_zy)
+                        
+                        # Boundary color and other options
+                        boundary_r.module_manager.scalar_lut_manager.lut.table = lut
+                        boundary_l.module_manager.scalar_lut_manager.lut.table = lut
+                        boundary_r.actor.property.lighting = False
+                        boundary_r.actor.property.shading = False
+                        boundary_l.actor.property.lighting = False
+                        boundary_l.actor.property.shading = False       
+
+                # Set viewing angle - For some unknown reason we must redefine the camera position each time.
+                # This is something to do with the boundaries being replaced each time.
+                mpf.view_position(fig, view, nx, ny, nz)
+                
+                if save_images == True:
+                    prefix = 'R1_'+str(R1) + '_' + mode + '_' + vis_mod_string + view + '_'# + '_norho_'
+                    mlab.savefig('D:\\my_work\\projects\\Asymmetric_slab\\Python\\visualisations\\3D_vis_animations\\'
+                                 + prefix + str(t_ind+1) + '.png')
```

Comment:
Actually, I misspoke, it will still fall over and die, but not because of the separators - 'D:' isn't a thing in Unix so to make that work you'll have to check for what operating system you're on (which I assume you can do with something in the `os` module) and choose one of two paths based on that. You should still use `os.path.join` though, because it's a bit more robust and slightly prettier.

Alternatively if you move your save folder to somewhere in your home folder 'C:\Users\matt\' or whatever it is on Windows, then you can use `os.path.expanduser`, which takes a path and expands the `~` character to the proper path for the home folder.